doc/howtos/rdtraining/sprinkles.rst

```diff
@@ -0,0 +1,512 @@
+    </form>
+
+It means that the ``description`` field is invisible when ``is_partner`` is ``False``. It is
+important to note that a field used in an ``attrs`` **must** be present in the view. If it
+should not be displayed to the user, we can use the ``invisible`` attribute to hide it.
+
+.. exercise:: Use of ``attrs``
+
+    - Make the garden area and orientation invisible in the ``estate.property`` form view when
+      there is no garden.
+    - Make the 'Accept' and 'Refuse' button invisible once the offer state is set.
+    - Do not allow adding an offer when the property state is 'Offer Accepted', 'Sold' or
+      'Canceled'. To do this, use the ``readonly`` ``attrs``.
+
+.. warning::
+
+    Using a (conditional) ``readonly`` attribute in the view can be useful to prevent encoding
+    errors, but keep in mind that it doesn't provide any level of security! There is no check done
+    server-side, therefore it's always possible to write on the field through RPC call.
+
+List
+----
+
+.. note::
+
+    **Goal**: at the end of this section, the property and offer list views have color decorations.
+    Moreover, offers and tags are editable directly in the list, and the availability date is
+    hidden by default.
+
+    .. image:: sprinkles/media/decoration.png
+      :align: center
+      :alt: List view with decorations and optional field
+
+    .. image:: sprinkles/media/editable_list.gif
+      :align: center
+      :alt: Editable list
+
+In case model has only a few fields, it can be useful to edit records directly through the list
+view and not open the form view. In the real estate example, there is no need to open a form view
+to add an offer or create a new tag. It can be achieved thanks to the ``editable`` attribute.
+
+.. exercise:: Make list views editable
+
+    Make the ``estate.property.offer`` and ``estate.property.tag`` list views editable.
+
+On the other hand, when a model has a lot of fields it can be tempting to add too many fields in the
+list view, making it unclear. An alternative method is to add the fields, but make them optionally
+hidden. It can be achieved thanks to the ``optional`` attribute.
+
+.. exercise:: Make field optional
+
+    Make the field ``date_availability`` on the ``estate.property`` list view optional and hidden by
+    default.
+
+Finally, color codes are useful to visually emphasize records. For example, in the real estate
+module we would like refused offers in red and the accepted offer in green. It can be achieved
+thanks to the ``decoration-{$name}`` attribute (see :ref:`reference/js/widgets` for a
+complete list):
+
+.. code-block:: xml
+
+    <tree decoration-success="is_partner==True">
+        <field name="name">
+        <field name="is_partner" invisible="1">
+    </tree>
+
+The records where ``is_partner`` is ``True`` will be displayed in green.
+
+.. exercise:: Add some decoration
+
+    On the ``estate.property`` list view:
+
+    - Properties with an offer received are green
```

Comment:
At this point, we still don't have the implementation of the state switch to "Offer Received" which is done in Inheritance. Neither is the implementation of "Offer Accepted".

Reply:
You can simply add the 'state' field to the form view, and change it by hand to test the different colours.

Maybe we add this as a 'tip'